routers/repo/repo.go

```diff
@@ -311,7 +311,7 @@ func Download(ctx *context.Context) {
 			ctx.Handle(500, "GetTagCommit", err)
 			return
 		}
-	} else if len(refName) == 40 {
+	} else if len(refName) > 7 && len(refName) <= 40 {
```

Comment:
nope, should be `7 <= len(refName) <= 40` just like it is, otherwise `f00badf00d.tar.gz` wouldn't work 🙂